src/evaluators/utility/PHAL_ComputeBasisFunctionsSide_Def.hpp

```diff
@@ -116,32 +118,199 @@ postRegistrationSetup(typename Traits::SetupData d,
   intrepidBasis->getValues(grad_at_cub_points, cub_points, Intrepid2::OPERATOR_GRAD);
 
   // BF does not depend on the current element, so we fill it now
-  std::vector<PHX::DataLayout::size_type> dims;
-  BF.fieldTag().dataLayout().dimensions(dims);
-  int numCells = dims[0];
-  for(int cell=0; cell<numCells; ++cell)
-  {
-    for (int side=0; side<numSides; ++side)
+  if (!useCollapsedSidesets) {
+    for(int cell=0; cell<numCells; ++cell)
+    {
+      for (int side=0; side<numSides; ++side)
+      {
+        for (int node=0; node<numSideNodes; ++node)
+        {
+          for (int qp=0; qp<numSideQPs; ++qp)
+          {
+            BF(cell,side,node,qp) = val_at_cub_points(node,qp);
+          }
+        }
+      }
+    }
+  } else {
+    for (int cellside = 0; cellside < BF.extent(0); ++cellside)
     {
       for (int node=0; node<numSideNodes; ++node)
       {
         for (int qp=0; qp<numSideQPs; ++qp)
         {
-          BF(cell,side,node,qp) = val_at_cub_points(node,qp);
+          BF(cellside,node,qp) = val_at_cub_points(node,qp);
         }
       }
     }
   }
 
-  cellsOnSides.resize(numSides);
-  numCellsOnSide.resize(numSides, 0);
-  for (int i=0; i<numSides; i++)
-    cellsOnSides[i] = Kokkos::DynRankView<int, PHX::Device>("cellOnSide_i", numCells);
+  // cellsOnSides.resize(numSides);
+  // numCellsOnSide.resize(numSides, 0);
+  // for (int i=0; i<numSides; i++)
+  //   cellsOnSides[i] = Kokkos::DynRankView<int, PHX::Device>("cellOnSide_i", numCells);
 
   d.fill_field_dependencies(this->dependentFields(),this->evaluatedFields());
   if (d.memoizer_active()) memoizer.enable_memoizer();
 }
 
+// *********************************************************************
+// Kokkos functor
+template<typename EvalT, typename Traits>
+KOKKOS_INLINE_FUNCTION
+void ComputeBasisFunctionsSide<EvalT, Traits>::
+operator() (const ComputeBasisFunctionsSide_Collapsed_Tag& tag, const int& sideSet_idx) const {
+
+  // Get the local data of side and cell
+  const int cell = sideSet.elem_LID(sideSet_idx);
+  const int side = sideSet.side_local_id(sideSet_idx);
+
+  // Computing tangents (the basis for the manifold)
+  for (int itan=0; itan<numSideDims; ++itan) {
+    for (int icoor=0; icoor<numCellDims; ++icoor) {
+      for (int qp=0; qp<numSideQPs; ++qp) {
+        tangents(sideSet_idx,qp,icoor,itan) = 0.;
```

Comment:
It might be beneficial to subview at least the sideSet_idx dimension.

We might even fuse all the qp loops into a big loop (and maybe add that dimension to the RangePolicy) and subview both sideSet_idx and qp?